uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/ContactWorkPattern.php

```diff
@@ -152,7 +152,7 @@ public static function getForDate($contactID, DateTime $date) {
     $contactWorkPatternTable = self::getTableName();
     $workPatternTable = WorkPattern::getTableName();
 
-    $query = "SELECT * FROM {$contactWorkPatternTable} cwp
+    $query = "SELECT cwp.* FROM {$contactWorkPatternTable} cwp
```

Comment:
Looks like `getForDate` and `getAllForPeriod` are doing almost the same thing. What if we change the former to only delegate the job to the latter? In that case, the "period" would be the date (i.e. the same value for both $start and $end)